tensorrt_llm/runtime/kv_cache_manager_v2/_storage_manager.py

```
@@ -727,6 +727,9 @@ def adjust_cache_level(
     def ratio_from_length(
         self, tokens_per_block: int, history_length: int, capacity: int
     ) -> TypedIndexList[PoolGroupIndex, float]:
+        if capacity < history_length:
+            warnings.warn("Bad sampling for capacity and history_length")
```

Ruff (0.15.12)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)